dist/dialog.linker.build.js

```diff
@@ -71,17 +78,21 @@ window.jQuery = window.jQuery || window.shoestring;
 		this.$background.attr( "id", this.$el.attr( "id" ) + "-background" );
 		this._setBackgroundTransparency();
 
-		if( this.isSetScrollPosition() ) {
-			this.scroll = "pageYOffset" in w ? w.pageYOffset : ( docElem.scrollY || docElem.scrollTop || ( body && body.scrollY ) || 0 );
-			this.$el[ 0 ].style.top = this.scroll + "px";
-		} else {
-			this.$el[ 0 ].style.top = '';
-		}
+		this.scroll = "pageYOffset" in w ? w.pageYOffset : ( docElem.scrollY || docElem.scrollTop || ( body && body.scrollY ) || 0 );
+		this.$el[ 0 ].style.top = this.scroll + "px";
 
 		$html.addClass( cl.open );
 		this.isOpen = true;
 
-		window.location.hash = this.hash;
+		var cleanHash = w.location.hash.replace( /^#/, "" );
+		var lastHash = w.location.hash.split( "#" ).pop();
+
+		if( cleanHash.indexOf( "-dialog" ) > -1 && lastHash !== this.hash ){
+			w.location.hash += "#" + this.hash;
```

Comment:
Just checking my understanding, if a dialog is open at `http://foo#bar-dialog` and a nested dialog is opened then it will be `http://foo#bar-dialog#baz-dialog`. Is there any concern about weird behavior when we concatenate hashes that way?